ruby/vampires.rb

```diff
@@ -0,0 +1,97 @@
+# Methods to check user input
+def check_if_integer
+  begin
+    num = Integer(gets.chomp)  
+  rescue 
+    puts 'Error reading input - please enter an integer number:'
+    retry
+  end
+
+  num
+end
+
+def yes_or_no
+  loop do
+    case gets.chomp.downcase
```

Comment:
I avoid using gets.chomp inside of methods.  Try passing this value i as a parameter instead.